third-culture-software/bhima
# Delete asset-related data from inventory item when it is not an asset

When the user changes an inventory item to an asset and adds asset-specific data (manufacturer brand, etc) that data is saved in the inventory item.  If the user decides that it should not be an asset, edits the inventory item, and changes it to NOT an asset, this PR fixes things so that the asset-related data is removed.

**QUESTION:**  Do we want manufacturer brand and model names for non-asset stock?  If so, this PR should be updated and only the reference_number should be removed and the edit form should be changed to not hide the manufacturer data when it is not an asset.

Closes https://github.com/IMA-WorldHealth/bhima/issues/6402

**TESTING**
- Use any data set
- Open Inventory > Inventory Registry
- In the [Menu] > Columns, display the is_asset, reference_number, manufacturer_brand, and manufacturer_model columns AND save the Column configuration.
- Using the action menu, edit an inventory item, change it to an asset, enter reference_number, manufacturer brand, etc, and save it.
- Notice the asset-related data you entered in the columns
- Edit the inventory item again and  change it to NOT an asset, save
- The asset-related data should now be gone.
-
